renku/infrastructure/repository.py

```diff
@@ -902,22 +931,32 @@ class SubmoduleManager:
 
     def __init__(self, repository: git.Repo):
         self._repository = repository
+        self._submodule_cache: Dict[git.Submodule, Submodule] = {}  # type: ignore
```

Comment:
Is `Submodule` object reused by GitPython? Maybe its better to use submodule's name as the key to be on the safe side.